### ISSUE TITLE
Users can get Initial Reputation from their pre-existing Citations

### ISSUE THREAD

**Issue by danielBingham** [danielBingham/peerreview#34]
As a user I want to have work I've already done in the field but not posted on Peer Review represented in my reputation so I can start participating on Peer Review at a level commensurate with my knowledge and experience.

## Story Background
*What is the history or background behind this story?  What context the developer researching or implementing this story need?*

We need a way to give experienced researchers initial reputation.  The easiest way to do this is to just treat citations as votes for the purposes of initial reputation.  There are a couple of potential places where we can get citation counts for potential researchers - Google Scholar being the most obvious one.  We should be able to either scrape a citation count or find an API where we can get it.  And then we can apply it to fields either by allowing researchers to divy it up in the way that most makes sense to them, or by scaping tagging information from somewhere.  

Either way, we want to allow researchers to initialize their reputation because that will kickstart the community.  This won't work with out it.

## Acceptance Criteria
*At what point is this story considered "done"?*
- Researchers can point us to a profile they have on another site (Google Scholar, etc) where we can scrape a citation count for them.
    - We treat each citation as an upvote.
- We have a way to assign the reputation gained from citations to fields.  Either the user can do it, or we do it for them through some sort of scraping.

## Dependencies
*What stories does this one depend on?  What do we need to do first before we can call this one done?*

**Comment by c-trimm** [comment]
I was looking into this a bit and was thinking of starting here, but have some questions below that might make this more complicated. 

Scraping shouldn't be too hard, but could prove unreliable if they change their classes or markup. There are APIs for it, but those simply do the scraping for you, but they might fix things faster if it breaks. One I found was SerpAPI, which can grab the citation count pretty easily, as well as papers, areas of interest, etc... that could help prepopulate some other things. https://serpapi.com/google-scholar-author-api -- only issue is that it's a paid service. They have a free plan for 100 searches/month, or $50/mo for 5000 searches, so we might be better off doing the scraping manually and fixing it if it breaks.

### Questions:
* How can we verify that they're entering a profile that belongs to them and not someone else's? I imagine we could display it on their PeerReview profile, and if someone catches that their GS profile doesn't match, they could report it and we could revoke their reputation. Moderation sucks, though, and you could get trolls giving false reports.
* I'm guessing if they change their Scholar profile, we'd need to revoke reputation and replace it based on the new profile, but that could allow someone to put in a false profile, get the rep, do some things on the site using that rep, then go change it so they don't get reported.
* Should we also implement a "rescan" button so that if they publish a new paper listed there, or gain more citations, it updates their citations/reputation?
* Do you have any other sites in mind beyond Google Scholar? If there are multiple sites, is there a chance of incorrectly double-counting citations for papers?

**Comment by danielBingham** [comment]
So I actually just found a lead for a database with an open API (that's open source) that we could get initial citation counts.  

https://docs.openalex.org/about-the-data/author#cited_by_count

I think we should investigate using that.  Also, sorry I'm only seeing this now - Github's notification system really sucks!  I need to figure out how to make sure I notice when there are comments on issues.

As for your other questions:

1. Verification is going to be a challenge.  We should look into using ORCID.  We can take their ORCHID on registration and use it to query OpenAlex.  If they don't give us an ORCID, then we could try querying by name.  OpenAlex mentions doing author normalization.  We could always make having an ORCID or Scholar profile a requirement of initalizing the reputation.
2. I was thinking of this as just being initialization, I don't think we should worry about keeping them in sync after that initial reputation boost.
3. See above.
4.  Yeah, OpenAlex is the best lead I have beyond Google Scholar.  I think we should just go with OpenAlex at first, and then we can investigate whether it makes sense to try to scrape Scholar as a backup.

**Comment by danielBingham** [comment]
Okay, this works.  It just needs to be wired into the ORCID iD connection process.

...It's really fucking cool seeing this working!

**Comment by danielBingham** [comment]
Remaining todo:

- [ ] Wire generating reputation from ORCID iD into the ORCID iD connection flow.
- [ ] Handle errors.

**Comment by danielBingham** [comment]
We may want to explore a message queue and workers for this one.  Generating the reputation takes a long time for users with a lot of publications.

**Comment by danielBingham** [comment]
We'll want to test this more in alpha testing, but it's all wired together now.  It's going to be very hard to test until we can find some folks with real ORCID iDs.
